ecosystem/sep-0040.md

```diff
@@ -0,0 +1,198 @@
+
+All other values for the `type` are considered non-standard until they are approved for inclusion within this SEP.
+
+##### `hash`
+The `hash` is provided to ensure integrity of the metadata that the account is pointing to. This prevents the data at the destination from being modified without explicit knowledge and approval by the issuer.
+
+If it is excluded, the NFT **must not be considered genuine**, _unless_ the `type` is `ipfshash`, since the IPFS CID includes an integrity check on the data it points to. 
+
+##### Storage
+Fields should be stored as simple key-value pairs within an account's data entries. The standard limitations of `ManageDataOp` apply: field names and their values can only be 64 bytes each. Most metadata attributes should be stored at the NFT reference location itself.
+
+**Backward Compatibility:** Existing NFT platforms essentially default to `type: ipfshash`. While this standard aims to free up the ecosystem from dependence on IPFS, it should also work with existing conventions that developed organically. To do so, **if you see an account with a lone `ipfshash` data entry, you may treat it as equivalent to an NFT account with `type: ipfshash`**.
+
+##### IPFS Storage
+If the `type` is `ipfshash`, the `ipfshash` key must point to an [IPFS Content IDentifier](https://docs.ipfs.io/concepts/content-addressing/) corresponding to the location of the metadata.
+
+If you are the issuer, make sure you understand the implications of using IPFS. You must use an IPFS lookup service to actually resolve to the metadata. For example, [this URL](https://cloudflare-ipfs.com/ipfs/QmaydGEshbL7HMQkD6FZPcMeEt6o15aC4p4P82VxQ7amjy) resolves to the Meridian 2021 Hat NFT via the Cloudflare IPFS service. If you're the issuer, you must also ensure that IPFS actually stores your data (by running your own node or entrusting a provider).
+
+##### URL Storage
+If the `type` is `uri`, the corresponding `uri` key must point to a [valid URI](https://datatracker.ietf.org/doc/html/rfc3986) containing the NFT's metadata. 
+
+_Note_: While you could technically describe metadata living on the IPFS with a URL by pointing to a specific node or a URI prefixing the CID with `ipfs://`, we recommend describing _only_ its CID to leave lookups to the users.
+
+#### Standard Metadata Specification
+This section describes the metadata itself rather than how to find it. 
+
+The metadata must be valid JSON and must contain the following **required** fields:
+
+| Field       | Description |
+| ----------- | ----------- |
+| `name` | a short name or title for the NFT |
+| `description` | a longer, more-detailed description of the NFT |
+| `type` | a [Media Type](https://www.iana.org/assignments/media-types/media-types.xhtml) describing the NFT's primary format (pointed to by `url`) |
+| `uri` **or** `url`  | a [URI](https://datatracker.ietf.org/doc/html/rfc3986) pointing to the actual NFT contents |
```

Comment:
Could we pick one field name, or do we need both for consistency with other standards?

Reply:
Well I want it to be `uri`, since that's broader, but Litemint already uses `url`, so I shoe-horned both in here. I'm not happy about it, though. We could say that the name is `url` but it can be any URI??

Reply:
A URI is just an identifier, a URL is a locator. Given that the intention here is to define a location that the NFT is stored, I think URL is fine and easily understood by a larger audience anyway.

There is probably not a huge amount of utility in saying it can be an identifier that isn't a locator, so I don't think URI is worth it.

Reply:
To be clear `ipfs://...` is a URL.

Reply:
URL. This is the way.

https://danielmiessler.com/study/difference-between-uri-url/

Reply:
That's a good point, I think I mixed myself up double-checking my statements against too many RFCs.